apps/explorer/src/components/module/PkgModulesWrapper.tsx

```diff
@@ -48,14 +52,51 @@ function PkgModuleViewWrapper({ id, modules }: Props) {
     const [searchParams, setSearchParams] = useSearchParams();
     const [query, setQuery] = useState('');
 
-    const paramModule = searchParams.get('module') || modulenames?.[0] || null;
-
-    const [selectedModule, setSelectedModule] = useState(
-        !!paramModule && modulenames.includes(paramModule)
-            ? paramModule
-            : modulenames[0]
+    const getModule = (
```

Comment:
I don't really think this needs to be a function defined here, you could just flatten this logic into the component itself. All of the variables you pass in are already available in scope here anyway.